create-project-async/src/graphql_document_creator.py

```diff
@@ -0,0 +1,98 @@
+import glob
+import os
+from concurrent.futures import ThreadPoolExecutor, as_completed
+from typing import TypedDict
+
+from requests import post
+
+
+class GraphQLDocument(TypedDict):
+    document: dict
+    extras: list[dict] | None
+
+
+class GraphQLDocumentCreator:
+    MAX_WORKERS = 8
+
+    def __init__(self, proxy_url: str, headers: dict[str, str], documents_path: str):
+        self.proxy_url = proxy_url
+        self.headers = headers
+        self.documents_path = documents_path
+
+    def create(self):
+        mapped_documents = self.__get_mapped_documents()
+        return self.__get_graphql_documents(mapped_documents)
+
+    def __get_mapped_documents(self):
+        filepaths = list(glob.iglob(f"{self.documents_path}/*"))
+        sorted_filepaths = self.__sort_possible_extra_files_last(filepaths)
+        mapped_documents = self.__map_documents(sorted_filepaths)
+        return mapped_documents
+
+    def __get_graphql_documents(self, mapped_documents: dict[str, dict]):
+        def upload_and_create_document(key: str):
```

Comment:
minor, and just curious

kalo `upload_and_create_document` nya dijadiin class method juga bisa kan no? @ivanm-gdp